### ISSUE TITLE
Slip and hyper operations interact to cause immutable values

### ISSUE THREAD

**Issue by SqrtNegInf** [rakudo/rakudo#2482]
## The Problem
An interaction between 'slip' and values being generated via hyper operators leads to complaint about an immutable value.

## Expected Behavior
Assignments should work.

## Actual Behavior
Fails with complaint about immutable X (where X is the type being processed, Int, Num, Real, etc)

## Steps to Reproduce
The 2nd statement generates the error 
```
say <1 2> »xx» 2; # ok
((1 1) (2 2))
say |<1 2> »xx» 2; # no good
Cannot modify an immutable IntStr (1)
```
This also fails:
`say 1 «*« |cis(1).reals x 2`

## Environment
This is Rakudo version 2018.10-119-g18ded174e built on MoarVM version 2018.10-75-g1bf132125
implementing Perl 6.d.

**Comment by lizmat** [comment]
Fixed with 78f642039f

**Comment by AlexDaniel** [comment]
This one was actually a regression, so it makes sense that the behavior was restored. A small bit of info but saves a lot of time when reviewing!
